flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/ClientTransportErrorHandlingTest.java

```diff
@@ -134,12 +133,10 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
 		}).when(rich[1]).onError(isA(LocalTransportException.class));
 
 		// First request is successful
-		ChannelFuture f = requestClient.requestSubpartition(new ResultPartitionID(), 0, rich[0], 0);
-		assertTrue(f.await().isSuccess());
+		requestClient.requestSubpartition(new ResultPartitionID(), 0, rich[0], 0);
 
 		// Second request is *not* successful
-		f = requestClient.requestSubpartition(new ResultPartitionID(), 0, rich[1], 0);
-		assertFalse(f.await().isSuccess());
```

Comment:
The `rich[1],onError` is actually verified above, then it would call `sync.countDown()`. Otherwise the test would fail in timeout.

In `PartitionRequestClient#requestSubpartition`, the `delayMs` is 0 then it would not schedule `tcpChannel.writeAndFlush(request)`. So I thought it wrote the request in the same thread before. After double checking this, the write is actually done in netty thread, so I would consider the race conditions here.

Reply:
I move the verify of `rich[0]` behind `sync.await` to confirm the future operations are done. Because the end of `sync.await` indicates the rich[1] is set on error, and the request of rich[0] which is in front of rich[1] should also be done.